lms/static/js/ccx/async_grade_book_download.js

```diff
@@ -0,0 +1,143 @@
+var define = window.define || RequireJS.define;
+
+define("async_grade_book_download",
+  [
+    'backbone',
+    'underscore',
+    'gettext',
+    'text!templates/ccx/grade_book_download.underscore'
+  ],
+  function (Backbone, _, gettext, gradeBookDownloadTemplate) {
+    'use strict';
+    var GradeBookDownloadView = Backbone.View.extend({
+
+      events: {
+        "click a#ccx-calculate-grades-csv" : "asyncPrepareGradeBook"
+      },
+
+      initialize: function(options) {
+        // set interval to fetch list of files that are ready for download.
+        this.gradeBookPrepareUrl = options.gradeBookPrepareUrl;
+        this.listPreparedGradeBookUrl = options.listPreparedGradeBookUrl;
+        this.showErrorMessage = false;
+        this.showSuccessMessage = false;
+        this.showReadyGradeBooks();
+      },
+
+      render: function () {
+        this.$el.html(_.template(gradeBookDownloadTemplate) ({
+          showErrorMessage: this.showErrorMessage,
+          showSuccessMessage: this.showSuccessMessage
+        }));
+        return this;
+      },
+
+      reloadDownloadList: function() {
+        // refresh list of grade books, ready for download.
+        var self = this;
+        // Fetch interval for grade book download request 20 sec.
+        var POLL_INTERVAL = 20000;
+        this.timer = setInterval(function() {
+          self.showReadyGradeBooks();
+        }, POLL_INTERVAL);
+      },
+
+      showReadyGradeBooks: function() {
+        // Get list of grade book files that are ready for download
+        var self = this;
+        $.ajax({
+          dataType: 'json',
+          url: self.listPreparedGradeBookUrl
+        }).done(function(data) {
+          if (data.downloads.length > 0) {
+            self.renderListOfReadyDownload(data.downloads);
```

Comment:
All of your `render` functions should get folded into the Underscore template. If you are rendering a template and then immediately mutating the result based on other data, you then you should instead make those modifications part of the template. Note that it will be much easier if you use Backbone models -- the template can get its information from the model or collection that the view is bound to.